File: pkg/codegen/testing/utils/testdata/kubernetes-pod-pp/nodejs/kubernetes-pod.ts
import * as pulumi from "@pulumi/pulumi";
import * as kubernetes from "@pulumi/kubernetes";

const bar = new kubernetes.core.v1.Pod("bar", {
    apiVersion: "v1",
    metadata: {
        namespace: "foo",
        name: "bar",
        labels: {
            "app.kubernetes.io/name": "cilium-agent",
            "app.kubernetes.io/part-of": "cilium",
            "k8s-app": "cilium",
        },
    },
    spec: {
        containers: [
            {
                name: "nginx",
                image: "nginx:1.14-alpine",
                ports: [{
                    containerPort: 80,
                }],
                resources: {
                    limits: {
                        memory: "20Mi",
                        cpu: "0.2",
                    },
                },
            },
            {
                name: "nginx2",
                image: "nginx:1.14-alpine",
                ports: [{
                    containerPort: 80,
                }],
                resources: {
                    limits: {
                        memory: "20Mi",
                        cpu: "0.2",
                    },
                },
            },
        ],
    },
});
// Test that we can assign from a constant without type errors
const kind = bar.kind;

File: pkg/codegen/dotnet/testdata/kubernetes-pod-pp/kubernetes-pod.cs
using System.Collections.Generic;
using System.Linq;
using Pulumi;
using Kubernetes = Pulumi.Kubernetes;

return await Deployment.RunAsync(() => 
{
    var bar = new Kubernetes.Core.V1.Pod("bar", new()
    {
        ApiVersion = "v1",
        Metadata = new Kubernetes.Types.Inputs.Meta.V1.ObjectMetaArgs
        {
            Namespace = "foo",
            Name = "bar",
            Labels = 
            {
                { "app.kubernetes.io/name", "cilium-agent" },
                { "app.kubernetes.io/part-of", "cilium" },
                { "k8s-app", "cilium" },
            },
        },
        Spec = new Kubernetes.Types.Inputs.Core.V1.PodSpecArgs
        {
            Containers = new[]
            {
                new Kubernetes.Types.Inputs.Core.V1.ContainerArgs
                {
                    Name = "nginx",
                    Image = "nginx:1.14-alpine",
                    Ports = new[]
                    {
                        new Kubernetes.Types.Inputs.Core.V1.ContainerPortArgs
                        {
                            ContainerPortValue = 80,
                        },
                    },
                    Resources = new Kubernetes.Types.Inputs.Core.V1.ResourceRequirementsArgs
                    {
                        Limits = 
                        {
                            { "memory", "20Mi" },
                            { "cpu", "0.2" },
                        },
                    },
                },
                new Kubernetes.Types.Inputs.Core.V1.ContainerArgs
                {
                    Name = "nginx2",
                    Image = "nginx:1.14-alpine",
                    Ports = new[]
                    {
                        new Kubernetes.Types.Inputs.Core.V1.ContainerPortArgs
                        {
                            ContainerPortValue = 80,
                        },
                    },
                    Resources = new Kubernetes.Types.Inputs.Core.V1.ResourceRequirementsArgs
                    {
                        Limits = 
                        {
                            { "memory", "20Mi" },
                            { "cpu", "0.2" },
                        },
                    },
                },
            },
        },
    });

    // Test that we can assign from a constant without type errors
    var kind = bar.Kind;

});

